apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/Sgv2RowsResourceImpl.java

```diff
@@ -184,26 +184,25 @@ public Response createRow(
       throw invalidPayloadException(e);
     }
 
-    queryWithTable(
-        keyspaceName,
-        tableName,
-        tableDef -> {
-          final ToProtoConverter toProtoConverter = findProtoConverter(tableDef);
-
-          try {
-            return buildAddRowQuery(keyspaceName, tableName, payloadMap, toProtoConverter);
-          } catch (IllegalArgumentException e) {
-            throw new WebApplicationException(
-                "Invalid path for row to create, problem: " + e.getMessage(), Status.BAD_REQUEST);
-          }
-        });
-
-    // apparently no useful data in ResultSet, we should simply return payload we got:
-    return Response.status(Status.CREATED).entity(payloadAsString).build();
+    return queryWithTableAsync(
+            keyspaceName,
+            tableName,
+            tableDef -> {
+              final ToProtoConverter toProtoConverter = findProtoConverter(tableDef);
+
+              try {
+                return buildAddRowQuery(keyspaceName, tableName, payloadMap, toProtoConverter);
+              } catch (IllegalArgumentException e) {
+                throw new WebApplicationException(
+                    "Invalid path for row to create, problem: " + e.getMessage(),
+                    Status.BAD_REQUEST);
+              }
+            })
+        .map(any -> RestResponse.status(Response.Status.CREATED, payloadAsString));
```

Comment:
Should we handle 'onFailure' on the _queryWithTableAsync_ Uni response, since it might throw `WebApplicationException` internally when the passed function is invoked?

Reply:
My assumption was that exceptions would be handled by general exception mappers like `WebApplicationExceptionMapper` so there would be usually no need to handle them in flow, unless there is specific need convert particular exception with more information available.

WDYT @ivansenic ?

Reply:
@tatu-at-datastax
I believe that's at the REST API method level exceptions, where the mapper will work.
(E.g. https://github.com/stargate/stargate/blob/6d691f80deb1808acc71f10a559e123a04eb22ed/apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/Sgv2RowsResourceImpl.java#L44)

But, in this case when we transform an Uni , the exception might be thrown and this will be propagated back all the way to the client as Uni with a failure, I believe. 
(https://github.com/stargate/stargate/blob/6d691f80deb1808acc71f10a559e123a04eb22ed/apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/RestResourceBase.java#L144)

Please see the gist I just created https://gist.github.com/kathirsvn/c0c74eee5e9f0f6637bad28416b5083e

When you run this test, you could see the RuntimeException (WebApplicationException) is not thrown by the server method  and instead, it's returned as Uni failure. And I believe, the Exception Mapper will only work when the exception is thrown by the server method.

Please correct me if my understanding is incorrect on the Exception Mapper is incorrect. I haven't tested that part specifically.

Reply:
@kathirsvn Ok: I do not really know `Uni` handling well enough to know either way. I'll have to think about this more next week. Sounds like you may be right wrt problem; but if so I'd ideally need a reproduction that goes through call, to verify behavior before and after fix.

Reply:
Exception mappers will take care of this.